actions/FgAction.js

```diff
@@ -0,0 +1,205 @@
+/* ***********************************************************************
+ * ADOBE CONFIDENTIAL
+ * ___________________
+ *
+ * Copyright 2023 Adobe
+ * All Rights Reserved.
+ *
+ * NOTICE: All information contained herein is, and remains
+ * the property of Adobe and its suppliers, if any. The intellectual
+ * and technical concepts contained herein are proprietary to Adobe
+ * and its suppliers and are protected by all applicable intellectual
+ * property laws, including trade secret and copyright laws.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Adobe.
+ ************************************************************************* */
+const openwhisk = require('openwhisk');
+const { getAioLogger, actInProgress } = require('./utils');
+const appConfig = require('./appConfig');
+const FgUser = require('./fgUser');
+const FgStatus = require('./fgStatus');
+
+const FG_PROOCESS_ACTION = 'fgProcessAction';
+const logger = getAioLogger();
+const BAD_REQUEST_SC = 400;
+const AUTH_FAILED_SC = 401;
+const GEN_ERROR_SC = 401;
```

Comment:
Should we set this to 500?

Reply:
Thank you @sukamat, As suggested updated to 500.